src/Reader.js

```diff
@@ -419,15 +412,22 @@ class Reader extends EventEmitter {
 					}
 				}
 
-				try {
-					this.pendingLoadAuthenticationKey[key] = this.loadAuthenticationKey(parseInt(keyNumber), key);
-					await this.pendingLoadAuthenticationKey[key];
-				} catch (err) {
-					throw new AuthenticationError('unable_to_load_key', 'Could not load authentication key into reader.', err);
-				}
+				// Store the authentication promise in case other blocks are in process of authentication
+				this.pendingLoadAuthenticationKey[key] = this.loadAuthenticationKey(parseInt(keyNumber), key);
 
 			}
 
+			try {
+				await this.pendingLoadAuthenticationKey[key];
+			} catch (err) {
+				throw new AuthenticationError('unable_to_load_key', 'Could not load authentication key into reader.', err);
+			}
+			finally {
```

Comment:
Using finally is great solution! 👍 Thank you very much. I didn't realize that my previously suggested fix won't work.